deployment/docker-dev/Dockerfile-dev

```diff
@@ -0,0 +1,51 @@
+#--------- Generic stuff all our Dockerfiles should start with so we get caching ------------
```

Comment:
Newer docker compose lets you explicitly specify which docker file to use to build. So I would prefer that you simply do this:

* rename docker-prod dir to simply 'docker'
* add ``Dockerfile-dev.yml`` to that directory
* update ``docker-compose.yml`` to reference the ``Dockerfile-dev.yml`` for the dev service description
* change the ``Dockerfile-dev.yml`` to add requirements-dev.txt (which you should rename)

I will explain more on FaceTime which the motivation is behind this request if anything is not clear.